### PR TITLE
Add large file test for Rubix

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java [hunk 1]
```diff
@@ -196,6 +207,71 @@ public void testCacheRead()
...
+            outputStream.write(randomData);
+        }
+
+        long beforeRemoteReads = getCacheStats().getRemoteReads();
+        long beforeCachedReads = getCacheStats().getCachedReads();
+
+        assertTrue(Arrays.equals(randomData, readFileBytes(cachingFileSystem.open(file))));
+
+        // stats are propagated asynchronously, wait for them
+        sleep(1000L);
```
Comment (losipiuk) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java:228]: Do we always have to wait 1s? Would polling for stats in shorter intervals (like 100ms) make sense?

File: presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java [hunk 2]
```diff
@@ -196,6 +207,71 @@ public void testCacheRead()
...
+        long intermittentRemoteReads = getCacheStats().getRemoteReads();
+        assertGreaterThan(intermittentRemoteReads, beforeRemoteReads);
+
+        assertTrue(Arrays.equals(randomData, readFileBytes(cachingFileSystem.open(file))));
+
+        // stats are propagated asynchronously, wait for them
+        sleep(1000L);
+        long afterCachedReads = getCacheStats().getCachedReads();
+        long afterRemoteReads = getCacheStats().getRemoteReads();
+        assertGreaterThan(afterCachedReads, beforeCachedReads);
```
Comment (losipiuk) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java:238]: `intermittent` and `after` name prefixes are poor. Use something more explicit. `firstReadCachedRead`, `secondReadCachedReads`?

File: presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java [hunk 3]
```diff
@@ -196,6 +207,71 @@ public void testCacheRead()
...
+                        }
+                    });
+        }
+        finally {
+            executorService.shutdownNow();
+        }
+
+        // stats are propagated asynchronously, wait for them
+        sleep(1000L);
+        assertGreaterThan(getCacheStats().getCachedReads(), afterCachedReads);
```
Comment (losipiuk) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java:271]: add comments to assertions so one does not have to reverse engineer the code.

File: presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java [hunk 4]
```diff
@@ -196,6 +207,71 @@ public void testCacheRead()
...
+        sleep(1000L);
+        long afterCachedReads = getCacheStats().getCachedReads();
+        long afterRemoteReads = getCacheStats().getRemoteReads();
+        assertGreaterThan(afterCachedReads, beforeCachedReads);
+        assertEquals(afterRemoteReads, intermittentRemoteReads);
+
+        // make sure parallel reading of large file works
+        ExecutorService executorService = newFixedThreadPool(10);
+        try {
+            List<Runnable> reads = nCopies(
```
Comment (losipiuk) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java:244]: `Callable`? No need for exception handling.

**Comment by sopel39** [comment]
AC @losipiuk

File: presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java (hunk 1 above)
Comment (losipiuk) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java:233]: With 100ms I meant polling in a loop. Not wait single interval.

**Pull request merged**